mvcc/backend/backend.go

```diff
@@ -358,7 +358,16 @@ func (b *backend) defrag() error {
 
 	b.batchTx.tx = nil
 
-	tmpdb, err := bolt.Open(b.db.Path()+".tmp", 0600, boltOpenOptions)
+	tmpdbPath := b.db.Path() + ".tmp"
```

Comment:
It'd be better to generate a unique name.

Reply:
I'll update the PR to do this and to do the deletion of any orphaned files on startup (since they can be large and we don't want them to accumulate).

Reply:
Switched to using `ioutil.TempFile`, which generates a unique name.